core/src/main/scala/kafka/log/Log.scala

```diff
@@ -893,7 +892,7 @@ class Log(@volatile private var _dir: File,
     // offset (see below). The next time the log is reloaded, we will load producer state using this snapshot
     // (or later snapshots). Otherwise, if there is no snapshot file, then we have to rebuild producer state
     // from the first segment.
-    if (messageFormatVersion < RecordBatch.MAGIC_VALUE_V2 ||
+    if (recordVersion.value < RecordBatch.MAGIC_VALUE_V2 ||
```

Comment:
How about replacing ```messageFormatVersion``` by ```recordVersion.value``` for all usages in this method.

Reply:
This was already done. Please let me now if thats not the case.